src/OrchardCore/OrchardCore.Deployment.Abstractions/DeploymentSourceBase.cs

```diff
@@ -0,0 +1,26 @@
+namespace OrchardCore.Deployment;
+
+/// <summary>
+/// Represents a base class for deployment source.
+/// </summary>
+/// <typeparam name="TDeploymentStep">The deployment step type.</typeparam>
+public abstract class DeploymentSourceBase<TDeploymentStep>
+    : IDeploymentSource<TDeploymentStep> where TDeploymentStep : DeploymentStep
```

Comment:
This abstract class should only implement `IDeploymentSource` It would have an abstract method like this and remove the new interface.

```
protected Task ProcessAsync(TDeploymentStep step, DeploymentPlanResult result);
```

Reply:
I already have this but I didn't commit it yet, and I thought you are the one who will ask for this :)